packages/synapse-react-client/src/components/SynapseChat/GridAgentChat.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react'
+import DraggableDialog from '../DraggableDialog/DraggableDialog'
+import { SynapseChat } from './index'
+import { ChatInteraction } from './SynapseChat'
+import { AgentSession } from '@sage-bionetworks/synapse-types'
+
+export type GridAgentChatProps = {
+  gridSessionId: string
+  usersReplicaId: number
+  chatbotName?: string
+  initialMessage?: string
+  open: boolean
+  onClose: () => void
+}
+
+export function GridAgentChat({
+  gridSessionId,
+  usersReplicaId,
+  chatbotName = 'Grid Assistant',
+  initialMessage,
+  open,
+  onClose,
+}: GridAgentChatProps) {
+  const [agentSession, setAgentSession] = useState<AgentSession | undefined>()
+  const [interactions, setInteractions] = useState<ChatInteraction[]>([])
+
+  // Create session context for grid sessions
+  const sessionContext = {
+    concreteType:
+      'org.sagebionetworks.repo.model.agent.GridAgentSessionContext' as const,
+    gridSessionId,
+    usersReplicaId,
+  }
```

Comment:
You can remove the `as const` by importing and using the type.

```typescript
import { AgentSession, GridAgentSessionContext } from '@sage-bionetworks/synapse-types'
```

```suggestion
  const sessionContext: GridAgentSessionContext = {
    concreteType:
      'org.sagebionetworks.repo.model.agent.GridAgentSessionContext',
    gridSessionId,
    usersReplicaId,
  }
```